lms/djangoapps/courseware/access.py

```diff
@@ -62,9 +63,41 @@
 )
 from courseware.access_utils import adjust_start_date, check_start_date, debug, ACCESS_GRANTED, ACCESS_DENIED
 
+from lms.djangoapps.ccx.models import CustomCourseForEdX
+
 log = logging.getLogger(__name__)
 
 
+def has_ccx_coach_role(user, course_key):
+    """
+    Check if user is a coach on this ccx.
+
+    Arguments:
+        user (User): the user whose descriptor access we are checking.
+        course_key (CCXLocator): Key to CCX.
+
+    Returns:
+        bool: whether user is a coach on this ccx or not.
+    """
+    # To avoid circular imports
+    from lms.djangoapps.ccx.views import CCXLocatorValidationException
+    if isinstance(course_key, CCXLocator):
+        ccx_id = course_key.ccx
+        role = CourseCcxCoachRole(course_key)
+
+        if role.has_user(user):
+            list_ccx = CustomCourseForEdX.objects.filter(
+                course_id=course_key.to_course_locator(),
+                coach=user
+            )
+            if list_ccx.exists():
+                coach_ccx = list_ccx[0]
+                return str(coach_ccx.id) == str(ccx_id)
+    else:
+        raise CCXLocatorValidationException
```

Comment:
Can you add a message to this exception, so that it's easier for a developer to debug? An exception message can make it much easier to understand *why* an exception has been raised.